src/preact/base-element.js

```diff
@@ -225,20 +228,29 @@ export class PreactBaseElement extends BaseElement {
   /**
    * Provides a mapping of Preact prop to AmpElement DOM attributes.
    *
-   * @protected {Object<string, AmpElementPropDef>}
+   * @type {Object<string, AmpElementProp>}
    */
   static props = {};
 
-  /** @param {Element} element */
+  /** @param {AmpElement} element */
```

Comment:
Would this clean up all the double-casts for this below?

```suggestion
  /**
   * @param {AmpElement} element
   * @type {typeof PreactBaseElement}
   */
```

Reply:
Testing this out!

Reply:
Nope :(